lib/mayaUsd/ufe/UsdUndoInsertChildCommand.cpp

```diff
@@ -164,7 +165,81 @@ UsdUndoInsertChildCommand::Ptr UsdUndoInsertChildCommand::create(
     return std::make_shared<MakeSharedEnabler<UsdUndoInsertChildCommand>>(parent, child, pos);
 }
 
-bool UsdUndoInsertChildCommand::insertChildRedo()
+static void doUsdInsertion(
+    const UsdStagePtr&    stage,
```

Comment:
This functions refactor the common code between undo and redo. The same refactor pattern is also used in other commands, like rename. So most of the following code is actually moved from below, with the enhancement required to deal with the session layers added..